### DISCUSSION TITLE
Is it possible to integrate with existing enterprise OIDC system?

### DISCUSSION THREAD

**Discussion by smartaquarius10** [ory/hydra#2510] [Q&A] [answered]
Hello,

We already have enterprise level OIDC setup. We are using it with other application like in C# or node js to enable single sign on functionality. 

Is it possible to integrate this enterprise open id using hydra package in golang. 

If possible please share some docs or url for the details. 

Thank you

**Comment by vinckr** [comment, marked as answer]
Yes is it.

Can you please provide some more details as to what enterprise system, is it a known provider or did you roll your own custom solution?

What does your use case look like, is it to just enable SSO for a web application ?

Before you jump into implementing Hydra, please do read this article carefully, it will help you decide if you need Hydra for your use case: https://www.ory.sh/hydra/docs/concepts/before-oauth2
If you are not building a system that involves third parties - for example a third party company (CircleCI) using your (GitHub) APIs - but instead are looking to solve "Single Sign On", mobile / single page app authentication, or generally "I want an API token after someone logs in", take a look at Ory Kratos instead!

Please note that there are a lot of different options.
Deciding the best course of implementation very much depends on what you want to achieve exactly, where are your performance "pain"points ect. 

Feel free to post more here or in our [chat](ory.sh/chat).

**Reply by smartaquarius10** [reply]
@vinckr , In the backend its ping federate. We have integrated this within .net core where we have to pass the values for client id, client secret etc. 

So my use case is I will open the website and via single sign-on it will get the data of the user at server side and from that data we will further perform few task. 

In short replacing old windows authentication features to oidc. 
Looking for something like [this](https://github.com/coreos/go-oidc) in Ory. 

I wasn’t aware of kratos. For my use case is Kratos enough. 

Thank you

**Reply by smartaquarius10** [reply]
@vinckr , Hello vincent,
Hope you’re doing well. 

Once you have any update on this please let me know as I’ve to integrate this within one of my go web applications.

**Reply by vinckr** [reply]
Hey, you dont have to wait, you can go ahead with the integration 🙂
I would reccommend you start with a rough PoC and when that works you can test it out in a production mockup. 
Best start with the [Quickstart](https://www.ory.sh/kratos/docs/next/quickstart/), the [overview](https://www.ory.sh/kratos/docs/next/concepts/index#introducing-ory-kratos) and [the Login flow,](https://www.ory.sh/kratos/docs/next/self-service/flows/user-login) then you can determine how best to extract the data you need.
We also have a [.NET client for Kratos](https://www.nuget.org/packages/Ory.Kratos.Client/).

**Reply by smartaquarius10** [reply]
@vinckr , but I want to it use as a client. Do we have to host Kratos separately as a service. I’m still not getting how to use it as client and integrate within existing go web application. I want to integrate my go application via oidc authentication flow using client id, secret id and redirect url etc. 

Is this possible with kratos. Do you have any documentation for this kind of use case in golang. Because in most of the docs curl request and js codes are written. 

Am I missing something here?

**Reply by Benehiko** [reply]
Hi @smartaquarius10 

Ory Kratos is for identity management which allows your users to register and sign in to your system. Kratos can integrate with an OIDC provider (e.g. sign in with Google). See here https://www.ory.sh/kratos/docs/guides/sign-in-with-github-google-facebook-linkedin

If you do not need to be the OIDC provider as well, then Kratos is enough for this. If you want to also be the OIDC provider, then you can use Ory Hydra.

**Reply by Benehiko** [reply]
Just to also clarify, Kratos will handle the OIDC flows for you, so you don't need any OIDC library in your application

**Reply by vinckr** [reply]
Hey @smartaquarius10,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
